src/plugins/intel_gpu/src/plugin/ops/gather.cpp

```diff
@@ -119,6 +119,7 @@ void CreateGatherOpBase(ProgramBuilder& p, const std::shared_ptr<T>& op, const i
                                         reordered_inputs[0],
                                         reordered_inputs[1],
                                         axis,
+                                        input_shape.to_shape(),
                                         out_shape,
                                         batch_dim,
```

Comment:
could you add unittest for this target usage?

Reply:
There  is the mixed_ranks_gather at src/plugins/intel_gpu/tests/unit/passes/reorder_inputs_test.cpp.
That's enough to check it.

Reply:
If the test was already there, why did it not catch the bug?

Reply:
At bad commit, the input/output rank were compared incorrectly as this old code.

src/plugins/intel_gpu/tests/unit/passes/reorder_inputs_test.cpp
```c++
TEST(reorder_inputs, mixed_ranks_gather) {
    // Topology:
    // (4d) -> conv -> (4d) -> border -> (4d) -> gather -> (5d) -> gather -> (6d) -> permute (6d)
    // In case when preferred format for convolution is selected as byxf (in the test it's enforced)
    // it could be propagated to border and gathers, but dimensions are handled incorrectly
    // and the second gather may complain that axis >= rank
    // So here we expect that input format for gather is aligned with actual output rank and format
...
    ASSERT_EQ(gather1_node.get_input_layouts()[0].format, format::bfzyx);   <<< 5d input. It should be 4d (bfyx)
    ASSERT_EQ(gather1_node.get_output_layout().format, format::bfzyx);      <<< 5d output

    ASSERT_EQ(gather2_node.get_input_layouts()[0].format, format::bfwzyx);  <<< 6d input. It should be 5d (bfzyx)
    ASSERT_EQ(gather2_node.get_output_layout().format, format::bfwzyx);     <<< 6d output
...
```